### ISSUE TITLE
styling hover,focus and active states differently

### ISSUE THREAD

**Issue by nirjan-dev** [nirjan-dev/learning-to-code#33]
[link to the article](https://zellwk.com/blog/style-hover-focus-active-states/)

Hover, focus, and active states should be styled different.

There’s a simple reason: They’re different states!

:hover triggers when a user brings their mouse over an element.

Hover states are usually represented by a change in background-color (and/or color). The difference in states doesn’t have to be obvious because users already know they hovered on something.

Styling focus states

:focus activates when an element receives focus. Elements can receive focus in two ways:

    When users tab into a focusable element
    When users click on a focusable element

Focusable elements are:

    Links (<a>)
    Buttons (<button>)
    Form elements (input, textarea, etc.)
    Elements with tabindex

Here are a few important points to note:

    Users cannot tab into an element with tabindex="-1", but they can click on it. The click triggers focus.
    On Safari and Firefox (Mac), clicks do not focus the <button> element. More info here.
    When you click on a link (<a>), focus remains on the link until you lift your finger from your mouse. When you lift your finger, the focus gets redirected elsewhere if the href points to a valid id on the same page.

Here are a few important points to note:

    Users cannot tab into an element with tabindex="-1", but they can click on it. The click triggers focus.
    On Safari and Firefox (Mac), clicks do not focus the <button> element. More info here.
    When you click on a link (<a>), focus remains on the link until you lift your finger from your mouse. When you lift your finger, the focus gets redirected elsewhere if the href points to a valid id on the same page.

You can use a combination of outline, border, and box-shadow properties to create nice focus styles. I share how to do this in “Creating a custom focus style”.

button {
  background-color: #dedede;
}

button:hover {
  background-color: #aaa;
}

button:focus {
  outline: none;
  box-shadow: 0 0 0 3px lightskyblue;
}

When you interact with things in real life, you expect some sort of feedback. For example, if you push a button, you expect the button to get pressed.

This feedback is useful on websites too. You can style the “push button” moment with :active. :active triggers when you interact with an element. Interacting here means:

    Holding down your left mouse button on an element (even non-focusable ones)
    Holding down the Space key (on buttons)

The magic combination allows users to get feedback when they hover, focus, and interact with an element. Here’s the code you need:

.element:hover,
.element:active {
  /* Change background/text color */ 
}

.element:focus {
  /* Show outline /* 
}

But if you think there’s not enough affordance, you’d want to style :hover, :focus, and :active separately.

.element:hover {
  /* Change background/text color */ 
}

.element:active {
  /* Another change in background/text color */
}

.element:focus {
  /* Show outline /* 
}
